pyvista/plotting/_plotting.py

```diff
@@ -0,0 +1,147 @@
+            feature_angle=feature_angle,
+        )
+        if scalars is not None:
+            ind = mesh.point_data['__orig_ids__']
+            scalars = np.asarray(scalars)[ind]
+    else:
+        mesh.compute_normals(cell_normals=False, inplace=True)
+
+    if texture:
+        mesh.active_t_coords = _tcoords
+
+    return mesh, scalars
+
+
+def process_opacity(mesh, opacity, preference, n_colors, scalars, use_transparency):
+    """Process opacity.
+
+    This function accepts an opacity string or array and always
+    returns an array that can be applied to a dataset for plotting.
+
+    Parameters
+    ----------
+    mesh : pyvista.DataSet
+        Dataset to process the opacity for.
+    opacity : str, numpy.ndarray
+        String or array.  If string, must be a cell or point data array.
+        preference : str, optional
+            When ``mesh.n_points == mesh.n_cells``, this parameter
+            sets how the scalars will be mapped to the mesh.  Default
+            ``'points'``, causes the scalars will be associated with
+            the mesh points.  Can be either ``'points'`` or
+            ``'cells'``.
+    n_colors : int, optional
+        Number of colors to use when displaying the opacity.
+    scalars : numpy.ndarray, optional
+        Dataset scalars.
+    use_transparency : bool, optional
+        Invert the opacity mappings and make the values correspond
+        to transparency.
+
+    Returns
+    -------
+    _custom_opac : bool
+        If using custom opacity.
+    opacity : numpy.ndarray
+        Array containing the opacity.
+
+    """
+    _custom_opac = False
+    if isinstance(opacity, str):
+        try:
+            # Get array from mesh
+            opacity = get_array(mesh, opacity,
+                                preference=preference, err=True)
+            if np.any(opacity > 1):
+                warnings.warn("Opacity scalars contain values over 1")
+            if np.any(opacity < 0):
+                warnings.warn("Opacity scalars contain values less than 0")
+            _custom_opac = True
+        except:
+            # Or get opacity transfer function
+            opacity = opacity_transfer_function(opacity, n_colors)
+        else:
+            if scalars.shape[0] != opacity.shape[0]:
+                raise ValueError(
+                    "Opacity array and scalars array must have the same number "
+                    "of elements."
+                )
+    elif isinstance(opacity, (np.ndarray, list, tuple)):
+        opacity = np.array(opacity)
+        if scalars.shape[0] == opacity.shape[0]:
+            # User could pass an array of opacities for every point/cell
+            _custom_opac = True
+        else:
+            opacity = opacity_transfer_function(opacity, n_colors)
+
+    if use_transparency and np.max(opacity) <= 1.0:
+        opacity = 1 - opacity
+    elif use_transparency and isinstance(opacity, np.ndarray):
+        opacity = 255 - opacity
```

Comment:
I was initially confused by this branch as we explicitly warn with opacities are >1.  When using `opacity_transfer_function`, it can be in the range 0-255.  Allowing multiple ranges makes every function really complex.  Can we enforce usage of one system or the other and then convert at point of use, if needed?

Reply:
> Can we enforce usage of one system or the other and then convert at point of use, if needed?

Yes, however I think this is out of the scope of this PR. When moving the opacity code into `_plotting.py`, we've discovered a variety of bugs as we're really giving this code a thorough review.  I've cleaned up the `use_transparency` branch a bit, but I think we should hold off any more changes to the opacity functions for another PR